GeneratorScript/sensor_data_generator.py

```diff
@@ -2,45 +2,142 @@
 import json
 import time
 import random
+import requests
+from datetime import datetime
+from concurrent.futures import ThreadPoolExecutor
+import threading
 
-RABBITMQ_HOST = 'localhost'  
-RABBITMQ_QUEUE = 'sensor_queue'
-RABBITMQ_USER = 'admin' 
-RABBITMQ_PASSWORD = 'admin' 
-
-SENSOR_NAMES = ["Temperature", "Humidity", "CO2", "UVIntensity"]
-SENSOR_RANGES = {"Temperature": (-20, 20), "Humidity": (0, 100), "CO2": (0, 2000), "UVIntensity": (0, 15)}
-
-def generate_sensor_data(sensor_id):
-    sensor_name = SENSOR_NAMES[sensor_id % len(SENSOR_NAMES)]
-    sensor_value = random.randint(SENSOR_RANGES[sensor_name][0], SENSOR_RANGES[sensor_name][1])
-    sensor_data = {
-        "name": sensor_name,
-        "number": sensor_id,
-        "value": sensor_value
-    }
-    return json.dumps(sensor_data)
-
-credentials = pika.PlainCredentials(username=RABBITMQ_USER, password=RABBITMQ_PASSWORD)
-connection_params = pika.ConnectionParameters(
-    host=RABBITMQ_HOST,
-    credentials=credentials
-)
-
-connection = pika.BlockingConnection(connection_params)
-channel = connection.channel()
-channel.queue_declare(queue=RABBITMQ_QUEUE)
-
-try:
-    while True:
-        for sensor_id in range(1, 17):
-            sensor_data = generate_sensor_data(sensor_id)
-            channel.basic_publish(exchange='',
-                                  routing_key=RABBITMQ_QUEUE,
-                                  body=sensor_data)
-            print(f"Sent: {sensor_data}")
-            time.sleep(1)
-except KeyboardInterrupt:
-    print("Stopping simulation...")
-finally:
-    connection.close()
+RABBITMQ_HOST = 'localhost'
+RABBITMQ_QUEUE = 'data_queue'
+RABBITMQ_USER = 'admin'
+RABBITMQ_PASSWORD = 'admin'
+
+API_BASE_URL = 'http://localhost:8000/api/sensors'
+
+SENSOR_TYPES = ["Temperature", "Humidity", "Pressure", "WindSpeed"]
+SENSORS_FOR_EACH_TYPE = 4
+
+SENSOR_RANGES = {
+    "Temperature": (-20, 50),
+    "Humidity": (0, 100),
+    "Pressure": (900, 1100),
+    "WindSpeed": (0, 150)
+}
+
+DATA_GENERATION_RATE = {
+    "Temperature": 5,
+    "Humidity": 2,
+    "Pressure": 3,
+    "WindSpeed": 4
+}
+
+stop_simulation = threading.Event()
+
+class Sensor:
+    def __init__(self, sensor_id=None, sensor_type=None, sensor_name=None):
+        self.sensor_id = sensor_id
+        self.sensor_type = sensor_type
+        self.sensor_name = sensor_name
```

Comment:
i think i don't get the difference between sensor_type and sensor_name. In the calculate_sleep_time you provide sensor type parameter, but then use DATA_GENERATION_RATE which has sensor_names in it. Could you confirm it should be really sensor_type nor sensor_name?